ui/test/nuxeo-document-layout.test.js

```diff
@@ -0,0 +1,314 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import '@webcomponents/html-imports/html-imports.min.js';
+import '../nuxeo-document-layout.js';
+import '../widgets/nuxeo-input.js';
+import '../widgets/nuxeo-textarea';
+import { Polymer } from '@polymer/polymer/polymer-legacy.js';
+import {
+  fixture,
+  flush,
+  isElementVisible,
+  html,
+  waitForEvent,
+  waitForAttrMutation,
+} from '@nuxeo/nuxeo-elements/test/test-helpers';
+import { LayoutBehavior } from '../nuxeo-layout-behavior.js';
+
+// Export Polymer and PolymerElement for 1.x and 2.x compat
+window.Polymer = Polymer;
+
+window.nuxeo.I18n.language = 'en';
+window.nuxeo.I18n.en = window.nuxeo.I18n.en || {};
+window.nuxeo.I18n.en['documentLayout.notFound'] = 'Failed to find {0} layout for {1}.';
+
+Nuxeo = Nuxeo || {};
+Nuxeo.LayoutBehavior = LayoutBehavior;
+
+// determine base module path (relies on @open-wc/webpack-import-meta-loader)
+const { url } = import.meta;
+const base = url.substring(0, url.lastIndexOf('/'));
+
+/* eslint-disable no-unused-expressions */
+suite('<nuxeo-document-layout>', () => {
+  let documentLayout;
+  const baseUrl = `${base}/layouts/document/`;
+
+  const buildDoc = (props) => {
+    const doc = { type: 'Test', uid: '12ae', properties: {} };
+    Object.assign(doc.properties, props);
+    return doc;
+  };
+
+  const buildLayout = async (doc = buildDoc(), layout = 'edit') => {
+    const dl = await fixture(
+      html`
+        <nuxeo-document-layout .document="${doc}" layout="${layout}" href-base="${baseUrl}"></nuxeo-document-layout>
+      `,
+    );
+    if (!dl.element) {
+      await Promise.race([
+        waitForEvent(dl.$.layout, 'element-changed'),
+        waitForAttrMutation(dl.$.layout.$.error, 'hidden', null),
+      ]);
+    }
+    await flush();
+    return dl;
+  };
+
+  const getWidget = (label) => {
+    if (!documentLayout || !documentLayout.element) {
+      return null;
+    }
+    return documentLayout.element.shadowRoot.querySelector(`[role="widget"][name="${label}"]`);
+  };
+
+  const layoutChangedListener = () =>
+    new Promise((resolve) => {
+      document.addEventListener('document-layout-changed', (e) => resolve(e));
+    });
+
+  const assertNotFound = () => {
+    expect(documentLayout.$.layout.$.error.hidden).to.be.false;
```

Comment:
can't we use `isElementVisible` here ?

Reply:
We are using it two lines below. Perhaps this is overtesting?